common/lib/common-utils/src/fluidError.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    IFluidError,
+    ITelemetryProperties,
+    ISensitiveDebugData,
+    isIFluidError,
+    ITelemetryBaseEvent,
+} from "@fluidframework/common-definitions";
+
+export abstract class FluidError extends Error implements IFluidError {
+    public abstract errorType: string;
+
+    private props: ITelemetryProperties = {};
+    private debugData: ISensitiveDebugData = {};
+    constructor(
+        message: string,
+        props: ITelemetryProperties = {},
+        debugData: ISensitiveDebugData = {},
+    ) {
+        super(message);
+        this.addDetails(props, debugData);
+    }
+
+    public getFluidTelemetryProps() { return { ...this.props, message: this.message, errorType: this.errorType }; }
+    public getSensitiveDebugData() {
+        return { ...this.debugData, stack: this.stack ?? "" };
```

Comment:
sholdn't the stack come from the inner error if it exists? otherwise this will loose the original call stack

Reply:
Yeah the "wrapAsFluidError" function, thanks for pointing out about preserving the original callstack.  I'll rethink it.

Reply:
Note that there are actually 3 places where stack info can come from - this.stack, this.innerError.stack and prepareErrorObject() can collect stack if it did not find in the error itself.